Description: Reproducible timestamp
 Use SOURCE_DATE_EPOCH as a timestamp to make the build reproducible.
Author: Eduard Sanou <dhole@openmailbox.org>

Index: hyperscan-4.2.0/CMakeLists.txt
===================================================================
--- hyperscan-4.2.0.orig/CMakeLists.txt
+++ hyperscan-4.2.0/CMakeLists.txt
@@ -6,7 +6,17 @@ set (HS_MINOR_VERSION 2)
 set (HS_PATCH_VERSION 0)
 set (HS_VERSION ${HS_MAJOR_VERSION}.${HS_MINOR_VERSION}.${HS_PATCH_VERSION})
 
-string (TIMESTAMP BUILD_DATE "%Y-%m-%d")
+if (DEFINED ENV{SOURCE_DATE_EPOCH})
+  execute_process(
+    COMMAND "date" "-u" "-d" "@$ENV{SOURCE_DATE_EPOCH}" "+%Y-%m-%d"
+    OUTPUT_VARIABLE BUILD_DATE 
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+else ()
+  execute_process(
+    COMMAND "date" "+%Y-%m-%d"
+    OUTPUT_VARIABLE BUILD_DATE
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+endif ()
 
 set(CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake)
 include(CheckCCompilerFlag)
